### DISCUSSION TITLE
Error converting images to black & white

### DISCUSSION THREAD

**Discussion by rael346** [ChenglongMa/SkinToneClassifier#14] [Q&A]
I'm encountering this error when using the `-bw` flag 
```
Error processing image [image path]: cannot reshape array of size 62500 into shape (3)
```
Not sure why this is the case, is my image size too big?

**Comment by ChenglongMa** [comment]
Hi @rael346,

Thanks for your valuable feedback!

I think it is a bug in my algorithm. Would you mind sharing this image so that I can debug it? (Sorry to keep asking for your data as I don't have a suitable dataset at hand to debug😥. I would appreciate your help!)

BTW, just confirm that `-bw` (`--black_white`) means:
> Whether to convert the input to black/white image(s). If true, the app will use the black/white palette to classify the image.

It will force convert images (even if they are in color) to **black and white** and then sort them with **black/white palette**.

If you just want to set the color or black/white palette to use without converting the image, you can use option `--image_type` or (`-t`).

Thanks.

**Reply by rael346** [reply]
HI @ChenglongMa 

Here is the image 

<img src="https://github.com/ChenglongMa/SkinToneClassifier/assets/75349763/9efbfc4d-4114-4d61-be4f-3f71a88fb37f" width="400" height="400"/>

Also I did try the `--image-type` option but when I ran it on the dataset that I'm using, it sometimes mis-classified the clothing as skin too, so I want to test if converting the images to black & white will help

**Comment by ChenglongMa** [comment]
Hi @rael346,

I have fixed the `-bw` error and released v1.1.2.
You can run 
```bash
pip install skin-tone-classifier --upgrade
```
to upgrade the library.

However, I tested the algorithms on the images you provided and found that this algorithm performs poorly a few AI generated images.
The algorithm cannot *effectively* identify **facial areas** and **skin areas** in the image.

This is because 
1. Sometimes **the distribution ratio of facial features** in the pictures generated by AI is incorrect; 
2. Furthermore, the algorithm is trained more on portraits taken by **cameras** under **natural light**, which are often different from the pictures generated by AI.
